osquery/sql/virtual_table.cpp

```diff
@@ -74,6 +73,194 @@ static inline std::string opString(unsigned char op) {
   return "?";
 }
 
+/// A list of tables that come from extensions; it is used to determine which
+/// table can be read/write
+std::vector<std::string> extension_table_list;
+Mutex extension_table_list_mutex;
+
+bool isExtensionTable(const std::string& table_name) {
+  ReadLock lock(extension_table_list_mutex);
+  return (std::find(extension_table_list.begin(),
+                    extension_table_list.end(),
+                    table_name) != extension_table_list.end());
+}
```

Comment:
0. There are 3 things that are used only together, would you like to make a make a class for them?
1. There is no usage of it outside of cpp file. Would you like to put it under anonymous namespace? 
2. As far as order of elements in `extension_table_list` is not important (I didn't find it, may be I'm wrong) I'd suggest to use `unordered_set` instead of it.